File: src/typescript/EnigmaSimulator.ts
const ALPHABET = "ABCDEFGHIJKLMNOPQRSTUVWXYZ";
const rotor: Map<string, string> = new Map<string, string>();

const shuffledAlphabet = "EKMFLGDQVZNTOWYHXUSPAIBRCJ"; // Rotor I configuration
for (let i = 0; i < ALPHABET.length; i++) {
    rotor.set(ALPHABET[i], shuffledAlphabet[i]);
}

function encrypt(input: string): string {
    return transform(input, rotor);
}

function decrypt(input: string): string {
    const reverseRotor: Map<string, string> = new Map<string, string>();
    rotor.forEach((value, key) => {
        reverseRotor.set(value, key);
    });
    return transform(input, reverseRotor);
}

function transform(input: string, mapping: Map<string, string>): string {
    let result = "";
    for (let char of input.toUpperCase()) {
        if (ALPHABET.includes(char)) {
            result += mapping.get(char);
        } else {
            result += char; // Non-alphabetic characters are not transformed
        }
    }
    return result;
}

// Test the implementation
const message = "HELLO WORLD";
const encryptedMessage = encrypt(message);
const decryptedMessage = decrypt(encryptedMessage);

console.log("Original Message: " + message);
console.log("Encrypted Message: " + encryptedMessage);
console.log("Decrypted Message: " + decryptedMessage);

File: src/java/EnigmaSimulator.java
package src.java;

import java.util.HashMap;
import java.util.Map;

public class EnigmaSimulator {
    private static final String ALPHABET = "ABCDEFGHIJKLMNOPQRSTUVWXYZ";
    private static final Map<Character, Character> rotor = new HashMap<>();

    static {
        String shuffledAlphabet = "EKMFLGDQVZNTOWYHXUSPAIBRCJ"; // Rotor I configuration
        for (int i = 0; i < ALPHABET.length(); i++) {
            rotor.put(ALPHABET.charAt(i), shuffledAlphabet.charAt(i));
        }
    }

    public static String encrypt(String input) {
        return transform(input, rotor);
    }

    public static String decrypt(String input) {
        Map<Character, Character> reverseRotor = new HashMap<>();
        for (Map.Entry<Character, Character> entry : rotor.entrySet()) {
            reverseRotor.put(entry.getValue(), entry.getKey());
        }
        return transform(input, reverseRotor);
    }

    private static String transform(String input, Map<Character, Character> mapping) {
        StringBuilder result = new StringBuilder();
        for (char c : input.toUpperCase().toCharArray()) {
            if (ALPHABET.indexOf(c) != -1) {
                result.append(mapping.get(c));
            } else {
                result.append(c); // Non-alphabetic characters are not transformed
            }
        }
        return result.toString();
    }
}
